### PR TITLE
Fix instances where nouns contain(s) or (s), for example app(s)

### PR DESCRIPTION
**Pull request by Luyao-Zhang-1**
Use regular expressions to replace all the instances in one batch:

Search (\w+)\(s\)
Replace $1 (or $1s)

### CODE REVIEW

File: content/en/docs/refguide/mobile/building-efficient-mobile-apps/offlinefirst-data/auto-committed-objects.md [hunk 1]
```diff
@@ -116,10 +116,10 @@ Auto-committed objects created during synchronization are deleted immediately af
 
 ```text {linenos=false}
 The offline synchronization detected {count} auto-committed objects during synchronization executed by {user}. Auto committed objects are not supported during offline synchronization. The Mendix runtime has deleted the following objects:
-- {Entity1}: {count} object(s) - ({guid1}, {guid2}, ... {guidN})
-- {Entity2}: {count} object(s) - ({guid1}, {guid2}, ... {guidN})
+- {Entity1}: {count} object (or objects) - ({guid1}, {guid2}, ... {guidN})
+- {Entity2}: {count} object (or objects) - ({guid1}, {guid2}, ... {guidN})
 ...
-- {EntityN}: {count} object(s) - ({guid1}, {guid2}, ... {guidN})
+- {EntityN}: {count} object (or objects) - ({guid1}, {guid2}, ... {guidN})
```
Comment (Yiyun333) [inline content/en/docs/refguide/mobile/building-efficient-mobile-apps/offlinefirst-data/auto-committed-objects.md]: Do we need to revert these changes too? They look like code but I am not very sure :) @ConnorLand

File: content/en/docs/developerportal/deploy/on-premises-design/ms-windows/_index.md [hunk 2]
```diff
@@ -206,7 +206,7 @@ Follow the instructions below and replace *[Name]* with the name of the rule in
 
 1. Open the IIS Manager and navigate to the website you want to manage.
 2. In the **Features View**, double-click **URL Rewrite**.
-3. In the **Actions** pane on the right side of the screen, click **Add rule(s)…** to add a new rewrite rule.
+3. In the **Actions** pane on the right side of the screen, click **Add rule (or rules)…** to add a new rewrite rule.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/on-premises-design/ms-windows/_index.md]:
```suggestion
3. In the **Actions** pane on the right side of the screen, click **Add rule(s)…** to add a new rewrite rule.
```

File: content/en/docs/developerportal/deploy/on-premises-design/ms-windows/_index.md [hunk 3]
```diff
@@ -235,7 +235,7 @@ This has to be the first rule; it is described after the rewrite rules to ensure
 4. Enter the **Server variable name** *HTTP_X_FORWARDED_PROTO*.
 5. Click **OK**.
 6. Click **Back to Rules**.
-7. Click **Add rule(s)…**.
+7. Click **Add rule (or rules)…**.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/on-premises-design/ms-windows/_index.md]:
```suggestion
7. Click **Add rule(s)…**.
```

File: content/en/docs/developerportal/deploy/on-premises-design/ms-windows/_index.md [hunk 4]
```diff
@@ -256,7 +256,7 @@ This has to be the first rule; it is described after the rewrite rules to ensure
 
 If HTTPS was configured at step 5.3 it is recommended to redirect all unencrypted HTTP traffic to HTTPS. To configure this, follow these steps:
 
-1. Click **Add rule(s)…**.
+1. Click **Add rule (or rules)…**.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/on-premises-design/ms-windows/_index.md]:
```suggestion
1. Click **Add rule(s)…**.
```

File: content/en/docs/refguide/modeling/domain-model/_index.md [hunk 5]
```diff
@@ -22,9 +22,9 @@ Below is a domain model that defines customers and orders. The names of the enti
 | --- | --- |
 | Annotation | A comment explaining an aspect of the domain model |
 | Entity Name | How the [entity](/refguide/entities/) will be referred to in the database |
-| Event Handler(s) | An indication that one or more [event handlers](/refguide/event-handlers/) have been set up for this entity |
+| Event Handler (or Handlers) | An indication that one or more [event handlers](/refguide/event-handlers/) have been set up for this entity |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/modeling/domain-model/_index.md]: This refers to what is in the associated image.
There isn't so much space to add the "or Handlers" so I've updated the image to remove the "(s)" altogether in this commit: ea8ded2eedaec614a4afb5e27051fc9cdaf9a657
```suggestion
| Event Handler | An indication that one or more [event handlers](/refguide/event-handlers/) have been set up for this entity |
```

File: content/en/docs/refguide/modeling/domain-model/_index.md [hunk 6]
```diff
@@ -22,9 +22,9 @@ Below is a domain model that defines customers and orders. The names of the enti
...
 | Image | An image which helps to identify the entity |
-| Validation Rule(s) | An indication that one or more [validation rules](/refguide/validation-rules/) have been set up for this attribute |
+| Validation Rule (or Rules) | An indication that one or more [validation rules](/refguide/validation-rules/) have been set up for this attribute |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/modeling/domain-model/_index.md]: Refers to image. See change in ea8ded2eedaec614a4afb5e27051fc9cdaf9a657.
```suggestion
| Validation Rule | An indication that one or more [validation rules](/refguide/validation-rules/) have been set up for this attribute |
```

File: content/en/docs/appstore/modules/saml.md [hunk 7]
```diff
@@ -488,7 +488,7 @@ If you need to change your identity provider metadata you can find more informat
...
 
-You can set up two sets of attributes, by adding new attributes, editing existing attributes, or removing selected attributes. These will be provided at different times. Those listed under **I want to request attribute(s) at my IDP during initial login** will be returned when the end-user initially signs in. Those listed under **I want to request attribute(s) at my IDP during in-session login** will be returned during [In-session Authentication](#in-session).
+You can set up two sets of attributes, by adding new attributes, editing existing attributes, or removing selected attributes. These will be provided at different times. Those listed under **I want to request attribute (or attributes) at my IDP during initial login** will be returned when the end-user initially signs in. Those listed under **I want to request attribute (or attributes) at my IDP during in-session login** will be returned during [In-session Authentication](#in-session).
```
Comment (MarkvanMents) [inline content/en/docs/appstore/modules/saml.md]: I think these are in the UX? @Karuna-Mendix can you check?
```suggestion
You can set up two sets of attributes, by adding new attributes, editing existing attributes, or removing selected attributes. These will be provided at different times. Those listed under **I want to request attribute(s) at my IDP during initial login** will be returned when the end-user initially signs in. Those listed under **I want to request attribute(s) at my IDP during in-session login** will be returned during [In-session Authentication](#in-session).
```

File: content/en/docs/developerportal/deploy/on-premises-design/ms-windows/sql-server/setting-up-a-sql-server-user.md [hunk 8]
```diff
@@ -26,9 +26,9 @@ To create a secure environment it is important to make a clear distinction betwe
     | `sysadmin` |  |  |  | **X** |
 
 * **Define the user mapping for the administrator:** Associate all the databases related to Mendix with the administrator. The template for these databases should be `dbo` and the role of the administrator should be: `db_owner`.
-* **Create (a) database user(s) for each Mendix application:** Each Mendix application should have its own database user(s).
+* **Create (a) database user (or users) for each Mendix application:** Each Mendix application should have its own database user (or users).
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/on-premises-design/ms-windows/sql-server/setting-up-a-sql-server-user.md]:
```suggestion
* **Create (a) database user(s) for each Mendix application:** Each Mendix application should have its own database user (or users).
```

File: content/en/docs/private-platform/configuration/pmp-configure-k8s.md [hunk 9]
```diff
@@ -248,7 +248,7 @@ Before creating any environments, you must register your Kubernetes clusters by
         ```
 
 3. Click **Save**.
-4. Click the newly created cluster and expand it, and then click **Retrieve Namespace(s)** to retrieve all the namespace and storage plans. 
+4. Click the newly created cluster and expand it, and then click **Retrieve Namespace (or Namespaces)** to retrieve all the namespace and storage plans.
```
Comment (MarkvanMents) [inline content/en/docs/private-platform/configuration/pmp-configure-k8s.md]: This looks like a UX element. @katarzyna-koltun-mx can you confirm?
```suggestion
4. Click the newly created cluster and expand it, and then click **Retrieve Namespace(s)** to retrieve all the namespace and storage plans. 
```

**Review by MarkvanMents — COMMENTED** [review]
Hi @Luyao-Zhang-1 
I've looked at the ones for Data and Runtime. I also spotted one in the Windows service console which I suspected hadn't been reviewed.

I've only looked at those on the assumption that other people will have proofread the rest.

But I did run this regex which, although it had some false positives, found a couple of examples which I think are in the UX `\*\*.*?\(or.*?s\).*?\*\*`. I have tagged the component owner to check.

Feel free to commit my suggestions if they make sense.

Thanks for doing this.

File: content/en/docs/appstore/modules/saml.md (hunk 7 above)
Comment (Karuna-Mendix) [reply content/en/docs/appstore/modules/saml.md]: Verified the UI text with the developer. It should be as it was earlier - **I want to request attribute(s) at my IDP during initial login**

**Comment by Karuna-Mendix** [comment]
Hi @Luyao-Zhang-1, I reviewed all IAM related docs (SAML, OIDC Provider, OIDC, and some other) for the changes and UI text. All are good.

**Comment by Luyao-Zhang-1** [comment]
> Hi @Luyao-Zhang-1, I reviewed all IAM related docs (SAML, OIDC Provider, OIDC, and some other) for the changes and UI text. All are good.

That' great. Thank you!

File: content/en/docs/private-platform/configuration/pmp-configure-k8s.md (hunk 9 above)
Comment (katarzyna-koltun-mx) [reply content/en/docs/private-platform/configuration/pmp-configure-k8s.md]: Correct, I believe it should be kept as-is.

**Comment by katarzyna-koltun-mx** [comment]
@Luyao-Zhang-1 I've reviewed the AWS, SAP, Private Cloud, and Private Platform docs. All good, minus 1 change that had already been suggested by Mark.

**Comment by Luyao-Zhang-1** [comment]
> @Luyao-Zhang-1 I've reviewed the AWS, SAP, Private Cloud, and Private Platform docs. All good, minus 1 change that had already been suggested by Mark.

Great, thank you @katarzyna-koltun-mx !

File: content/en/docs/refguide/mobile/building-efficient-mobile-apps/offlinefirst-data/auto-committed-objects.md (hunk 1 above)
Comment (ConnorLand) [reply content/en/docs/refguide/mobile/building-efficient-mobile-apps/offlinefirst-data/auto-committed-objects.md]: Yes, this is in a **```text** code snippet. Please revert all changes made inside code snippets.

File: content/en/docs/refguide/mobile/building-efficient-mobile-apps/offlinefirst-data/auto-committed-objects.md [hunk 10]
```diff
@@ -97,10 +97,10 @@ This is how the message will appear:
 
 ```text {linenos=false}
 The offline synchronization detected {count} auto-committed objects during synchronization executed by {user}. Auto committed objects are not supported during offline synchronization. The Mendix runtime has deleted the following objects:
-- {Entity1}: {count} object(s) - ({guid1}, {guid2}, ... {guidN})
-- {Entity2}: {count} object(s) - ({guid1}, {guid2}, ... {guidN})
+- {Entity1}: {count} object (or objects) - ({guid1}, {guid2}, ... {guidN})
```
Comment (ConnorLand) [inline content/en/docs/refguide/mobile/building-efficient-mobile-apps/offlinefirst-data/auto-committed-objects.md]: Revert these as well.

**Comment by Luyao-Zhang-1** [comment]
> Hi @Luyao-Zhang-1 I've looked at the ones for Data and Runtime. I also spotted one in the Windows service console which I suspected hadn't been reviewed.
> 
> I've only looked at those on the assumption that other people will have proofread the rest.
> 
> But I did run this regex which, although it had some false positives, found a couple of examples which I think are in the UX `\*\*.*?\(or.*?s\).*?\*\*`. I have tagged the component owner to check.
> 
> Feel free to commit my suggestions if they make sense.
> 
> Thanks for doing this.

Looks good! Thank you, @MarkvanMents , for checking the docs, especially for checking the UX text.
